platforms/software/build-init/src/main/java/org/gradle/buildinit/tasks/InitBuild.java

```diff
@@ -264,13 +271,40 @@ public void setupProjectLayout() {
             testFramework,
             insecureProtocol.get(),
             projectDir,
-            toolChainVersion);
+            toolChainVersion
+        );
+
+        boolean userInterrupted = inputHandler.interrupted();
+        if (userInterrupted) {
+            throw new BuildCancelledException();
+        }
+
         initDescriptor.generate(settings);
+        generateWrapper();
 
         initDescriptor.getFurtherReading(settings)
             .ifPresent(link -> getLogger().lifecycle(link));
     }
 
+    private void generateWrapper() {
+        FileResolver fileResolver = getFileOperations().getFileResolver();
+        File unixScript = fileResolver.resolve(WrapperDefaults.SCRIPT_PATH);
+        File jarFile = fileResolver.resolve(WrapperDefaults.JAR_FILE_PATH);
```

Comment:
🤔 This might be the same as looking at `ProjectLayout.projectDirectory.file(...)`?

Reply:
Using layout seems to require unnecessary boilerplate. What about `getProject().file(...)` ?

Reply:
Right, it's not compatible with CC...